raft/src/main/java/org/apache/kafka/raft/VotedState.java

```diff
@@ -92,11 +92,6 @@ public boolean hasElectionTimeoutExpired(long currentTimeMs) {
         return electionTimer.isExpired();
     }
 
-    public void overrideElectionTimeout(long currentTimeMs, long timeoutMs) {
```

Comment:
I understand that it is an unused method but removing or adding public methods requires a KIP in the Kafka community.

Reply:
I will revert this method deletion and let only these test cases. Sounds good? I wonder if adding a test case for such an (un-used) method would be beneficial...

Reply:
`VotedState` is not part of the public API so we can change this public method without a KIP.

Reply:
> ok, I stand corrected. Apologies for the confusion here @see-quick. (note to myself) Here is the guidance that we can follow for what requires a KIP for future: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals#KafkaImprovementProposals-Whatisconsidereda%22majorchange%22thatneedsaKIP?

Thanks for sharing that link @divijvaidya <3

Reply:
You can find the list of public classes/interfaces in the javadocs: https://kafka.apache.org/32/javadoc/allclasses-index.html

Reply:
Thanks noted.

Reply:
So basically we can merge it or? :) ^^ @divijvaidya ?

Reply:
Yes code looks good from my side 👍

We need to re-run the tests since the build had a bunch of failures due to a lingering thread. We expect this commit https://github.com/apache/kafka/commit/09011da76d450f9a7dd11e159741f8e69cca115b to mitigate the lingering thread problem and unblock the build. Please check if you have this commit in your repo. If not, I would suggest to rebase this test with `trunk` and run again. If you already have this commit, then we need to ask a committer to re-run the tests.

Reply:
I have got that commit in my local repo, so as you propose we need to ask a committer to re-run the tests :)